### ISSUE TITLE
Collection.watch() error Aborted on React Native SDK

### ISSUE THREAD

**Issue by BondPiyapan** [realm/realm-js#6174]
### How frequently does the bug occur?

Always

### Description

I’m need get data change realtime 

use collection.watch() return error [AbortError: Aborted]

use by Realm react-native

this code

  const user = useUser()
  useEffect(() => {
    const setupChangeStream = async () => {
      try {
        const collection = user!
          .mongoClient('mongodb-atlas')
          .db('dev')
          .collection('Job')

        for await (const change of collection.watch()) {
          // The change event will always represent a newly inserted perennial
          const { documentKey, fullDocument } = change
          console.log(`new document: ${documentKey}`, fullDocument)
        }
      } catch (error) {
        console.error(error)
      }
    }

    setupChangeStream()
  }, [user])

I have set accordingly Examples of every step finished.

[https://www.mongodb.com/docs/realm/sdk/react-native/app-services/query-mongodb/#real-time-change-notifications](url)

Thank you.

### Stacktrace & log output

```shell
[AbortError: Aborted]
```

### Can you reproduce the bug?

Always

### Version

12.1.0

### What services are you using?

Atlas Device Sync

### Are you using encryption?

Yes

### Platform OS and version(s)

All platform

### Build environment

Which debugger for React Native: ..

**Comment by maichongju** [comment]
I am using `expo` for my app. There is no `--variant=release` option, and I still get the `Abort` error when I use the `prod` mode. We need to find some workaround for this since this is the only way we can monitor changes from the database.

**Comment by elle-j** [comment]
You can always use [Device Sync](https://www.mongodb.com/docs/realm/sdk/react-native/sync-data/) and add listeners to [react to changes](https://www.mongodb.com/docs/realm/sdk/react-native/react-to-changes/). Can I ask what the reason is for not using Device Sync for your use case?

**Comment by maichongju** [comment]
Good point. I am not sure why I have not tried that feature yet. I will give it a try.

**Comment by maichongju** [comment]
After some review and thinking. I think there might be some difficulty for me to use Device Sync. I have a database that many collection and I only want to monitor the notification collection. As what I understand at the moment, Device Sync does not allow sync a single collection. Which means, all other data on the database will also be sync to the device which I don't want and need.

**Comment by elle-j** [comment]
Device Sync is very flexible with what should be synced to the device. In your case, to sync only a specific collection, you would only subscribe to that collection. Take e.g. the following example from the [docs](https://www.mongodb.com/docs/realm/sdk/react-native/sync-data/configure-a-synced-realm/#configure-a-synced-realm) where only objects of the `YourObjectModel` is subscribed to (in the `initialSubscriptions` field):
```ts
import React from 'react';
import {AppProvider, UserProvider, RealmProvider} from '@realm/react';

function AppWrapperSync() {
  return (
    <AppProvider id={APP_ID}>
      <UserProvider fallback={LogIn}>
        <RealmProvider
          schema={[YourObjectModel]}
          sync={{
            flexible: true,
            initialSubscriptions: {
              update(subs, realm) {
                subs.add(realm.objects(YourObjectModel));
              },
            },
          }}>
          <RestOfApp />
        </RealmProvider>
      </UserProvider>
    </AppProvider>
  );
}
```

Since you only have a subscription to `YourObjectModel`, no other collections will be synced.

If it would help, you can also have a look at one of our [example apps](https://github.com/realm/realm-js/tree/main/examples/rn-todo-list) and the provided README.

If you encounter what you think is a bug, feel free to open a new issue specifically for that.
